mlir/tools/mlir-rocm-runner/mlir-rocm-runner.cpp

```diff
@@ -34,6 +34,18 @@
 using namespace mlir;
 using namespace llvm;
 
+static cl::OptionCategory optFlags{"ROCm runner opt-like flags"};
+
+// CLI variables for -On options.
+static cl::opt<bool> optO0{
+    "Og0", cl::desc("Run GPU opt passes and codegen at O0"), cl::cat(optFlags)};
+static cl::opt<bool> optO1{
+    "Og1", cl::desc("Run GPU opt passes and codegen at O1"), cl::cat(optFlags)};
+static cl::opt<bool> optO2{
+    "Og2", cl::desc("Run GPU opt passes and codegen at O2"), cl::cat(optFlags)};
+static cl::opt<bool> optO3{
+    "Og3", cl::desc("Run GPU opt passes and codegen at O3"), cl::cat(optFlags)};
```

Comment:
why not make it an `cl::opt<int>` and the user can set an integer optimization level? I think your intention here is to be extra safe so user don't pick a crazy number, but we can have the logic to default to O3 if the user input doesn't make sense.

Reply:
Because
1) That's what's going on in the JitRunner
2) Paranoia good, silent defaults bad

Reply:
Well my aspect of paranoia is that you allow user to specify:
 `-optO0 -optO1 -optO2 -optO3`
at the same time, which is technically also incorrect.

Reply:
Fixed in the rebased version